lib/ecto/repo.ex

```diff
@@ -654,7 +654,8 @@ defmodule Ecto.Repo do
       end)
 
   """
-  @callback transaction(Keyword.t, fun) :: {:ok, any} | {:error, any}
+  @callback transaction(fun | Ecto.Multi.t, Keyword.t) ::
+    {:ok, any} | {:error, any} | {:error, atom, any, %{atom => any}}
```

Comment:
I don't like that we can have so different arguments and return values. Should we make a different function that would run a `Multi`?